src/nlp.jl

```diff
@@ -32,20 +53,52 @@ function getDual(c::ConstraintRef{NonlinearConstraint})
     return nldata.nlconstrDuals[c.idx]
 end
 
+type FunctionStorage
+    nd::Vector{NodeData}
+    adj::SparseMatrixCSC{Bool,Int}
```

Comment:
I haven't tested if this works with the ``sparse()`` constructor, but the storage for the ``Vector{Bool}`` for ``nzval`` is already 1/8th of the storage for the ``Vector{Int}`` for ``rowval``.